api/accounts/accounts.go

```diff
@@ -171,11 +172,11 @@ func (a *Accounts) handleCallContract(w http.ResponseWriter, req *http.Request)
 	if err := utils.ParseJSON(req.Body, &callData); err != nil {
 		return utils.BadRequest(errors.WithMessage(err, "body"))
 	}
-	revision, err := utils.ParseRevision(req.URL.Query().Get("revision"))
+	revision, err := utils.ParseRevision(req.URL.Query().Get("revision"), true)
 	if err != nil {
 		return utils.BadRequest(errors.WithMessage(err, "revision"))
 	}
-	summary, err := utils.GetSummary(revision, a.repo, a.bft)
+	header, st, err := utils.GetHeaderAndState(revision, a.repo, a.bft, a.stater)
```

Comment:
Should the GetHeaderAndState replace the GetSummary ? Or super seed it ?

We're using GetHeaderAndState in some places and GetSummary in others, it does seem to be possible to consolidate that behaviour into one ?

Reply:
Ya it seems `GetSummary` is only used now by `GetHeaderAndState`